packages/react-native-nitro-test/nitrogen/generated/shared/c++/Car.hpp

```diff
@@ -52,31 +52,29 @@ namespace margelo::nitro::test {
 
 namespace margelo::nitro {
 
-  using namespace margelo::nitro::test;
-
   // C++ Car <> JS Car (object)
   template <>
-  struct JSIConverter<Car> final {
-    static inline Car fromJSI(jsi::Runtime& runtime, const jsi::Value& arg) {
+  struct JSIConverter<margelo::nitro::test::Car> final {
+    static inline margelo::nitro::test::Car fromJSI(jsi::Runtime& runtime, const jsi::Value& arg) {
       jsi::Object obj = arg.asObject(runtime);
-      return Car(
+      return margelo::nitro::test::Car(
         JSIConverter<double>::fromJSI(runtime, obj.getProperty(runtime, "year")),
         JSIConverter<std::string>::fromJSI(runtime, obj.getProperty(runtime, "make")),
         JSIConverter<std::string>::fromJSI(runtime, obj.getProperty(runtime, "model")),
         JSIConverter<double>::fromJSI(runtime, obj.getProperty(runtime, "power")),
-        JSIConverter<Powertrain>::fromJSI(runtime, obj.getProperty(runtime, "powertrain")),
-        JSIConverter<std::optional<Person>>::fromJSI(runtime, obj.getProperty(runtime, "driver")),
+        JSIConverter<margelo::nitro::test::Powertrain>::fromJSI(runtime, obj.getProperty(runtime, "powertrain")),
+        JSIConverter<std::optional<margelo::nitro::test::Person>>::fromJSI(runtime, obj.getProperty(runtime, "driver")),
```

Comment:
This is what I was talking about @KrzysztofMoch - technically a struct can have nested properties where the name is also duplicate.
So now we use the fully qualified name, and got rid of the `using namespace` directive above - this now avoids duplicate definition errors! :)